php-frontend/src/main/java/org/sonar/php/cache/SymbolTableDeserializer.java

```diff
@@ -70,19 +75,20 @@ private ProjectSymbolData convert() {
       int sizeOfClassSymbols = readInt();
       for (int i = 0; i < sizeOfClassSymbols; i++) {
         ClassSymbolData data = readClassSymbolData();
-        projectSymbolData.add(data);
+        classSymbolData.add(data);
       }
       int sizeOfFuncSymbols = readInt();
       for (int i = 0; i < sizeOfFuncSymbols; i++) {
         FunctionSymbolData data = readFunctionSymbolDataList();
-        projectSymbolData.add(data);
+        functionSymbolData.add(data);
       }
       if (!"END".equals(in.readUTF())) {
         throw new IOException("Can't read data from cache, format corrupted");
       }
-      return projectSymbolData;
+      return SymbolTableImpl.create(classSymbolData, functionSymbolData);
```

Comment:
If we create `SymbolTableImpl` that way then fields in the class like `symbols`, `scopes` `symbolsByTree`, and `symbolByQualifiedName` are empty. Is it ok for the rest of the execution?